x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -61,3 +62,59 @@ func (qs queryServer) GetNetworkInferencesAtBlock(ctx context.Context, req *type
 
 	return &types.QueryNetworkInferencesAtBlockResponse{NetworkInferences: networkInferences}, nil
 }
+
+// Return full set of inferences in I_i from the chain, as well as weights and forecast implied inferences
+func (qs queryServer) GetLatestNetworkInference(
+	ctx context.Context,
+	req *types.QueryLatestNetworkInferencesAtBlockRequest,
+) (
+	*types.QueryLatestNetworkInferencesAtBlockResponse,
+	error,
+) {
+	topic, err := qs.k.GetTopic(ctx, req.TopicId)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "topic %v not found", req.TopicId)
+	}
+	if topic.EpochLastEnded == 0 {
+		return nil, status.Errorf(codes.NotFound, "network inference not available for topic %v", req.TopicId)
+	}
+	if topic.EpochLength == 0 {
+		return nil, status.Errorf(codes.NotFound, "block height last reward not available for topic %v", req.TopicId)
+	}
+
+	networkInferences, forecastImpliedInferenceByWorker, infererWeights, forecasterWeights, err := synth.GetNetworkInferencesAtBlock(
+		sdk.UnwrapSDKContext(ctx),
+		qs.k,
+		req.TopicId,
+		topic.EpochLastEnded,
+		topic.EpochLastEnded-topic.EpochLength,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryLatestNetworkInferencesAtBlockResponse{
+		NetworkInferences:         networkInferences,
+		InfererWeights:            qs.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:         qs.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences: qs.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+	}, nil
+}
+
+func (qs queryServer) ConvertWeightsToArrays(weights map[inference_synthesis.Worker]inference_synthesis.Weight) []*types.RegretInformedWeight {
+	weightsArray := make([]*types.RegretInformedWeight, 0)
+	for worker, weight := range weights {
+		weightsArray = append(weightsArray, &types.RegretInformedWeight{Worker: worker, Weight: weight})
+	}
+	return weightsArray
+}
+
+func (qs queryServer) ConvertForecastImpliedInferencesToArrays(
+	forecastImpliedInferenceByWorker map[string]*types.Inference,
+) []*types.WorkerAttributedValue {
+	forecastImpliedInferences := make([]*types.WorkerAttributedValue, 0)
+	for worker, inference := range forecastImpliedInferenceByWorker {
+		forecastImpliedInferences = append(forecastImpliedInferences, &types.WorkerAttributedValue{Worker: worker, Value: inference.Value})
+	}
+	return forecastImpliedInferences
+}
```

Comment:
Since they are utility functions, they don't need to be part of the `queryServer`. Making these functions standalone will improve reusability.